src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs

```diff
@@ -331,47 +331,56 @@ internal static List<CompletionResult> MakeCommandsUnique(IEnumerable<PSObject>
             List<CompletionResult> endResults = null;
             foreach (var keyValuePair in commandTable)
             {
-                var commandList = keyValuePair.Value as List<object>;
-                if (commandList != null)
+                if (keyValuePair.Value is List<object> commandList)
                 {
-                    endResults ??= new List<CompletionResult>();
-
-                    // The first command might be an un-prefixed commandInfo that we get by importing a module with the -Prefix parameter,
-                    // in that case, we should add the module name qualification because if the module is not in the module path, calling
-                    // 'Get-Foo' directly doesn't work
-                    string completionName = keyValuePair.Key;
-                    if (!includeModulePrefix)
+                    var modulesWithCommand = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+                    var importedModules = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+                    var commandInfoArray = new CommandInfo[commandList.Count];
+                    for (int i = 0; i < commandInfoArray.Length; i++)
                     {
-                        var commandInfo = commandList[0] as CommandInfo;
-                        if (commandInfo != null && !string.IsNullOrEmpty(commandInfo.Prefix))
+                        var commandInfo = (CommandInfo)commandList[i];
+                        commandInfoArray[i] = commandInfo;
+                        if (commandInfo.CommandType == CommandTypes.Application)
                         {
-                            Diagnostics.Assert(!string.IsNullOrEmpty(commandInfo.ModuleName), "the module name should exist if commandInfo.Prefix is not an empty string");
-                            if (!ModuleCmdletBase.IsPrefixedCommand(commandInfo))
-                            {
-                                completionName = commandInfo.ModuleName + "\\" + completionName;
-                            }
+                            continue;
                         }
-                    }
 
-                    results.Add(GetCommandNameCompletionResult(completionName, commandList[0], addAmpersandIfNecessary, quote));
+                        _ = modulesWithCommand.Add(commandInfo.ModuleName);
```

Comment:
Please remove the `_ =` part from below, I believe it's not needed.
```suggestion
                        modulesWithCommand.Add(commandInfo.ModuleName);
```